File: daily/cpp/1758_min_changes_to_make_alter_binary_string.cpp
#include <bits/stdc++.h>
using namespace std;
class Solution {
public:
    int minOperations(string s) {
        // int n = s.size();
        // int res = 0;
        // for (int i = 0; i < n; ++i) {
        //     if (s[i] - '0' != i % 2) {
        //         res += 1;
        //     }
        // }
        // return min(res, n - res);


        int cnt0 = 0, cnt1 = 0;
        for (int i = 0; i < s.size(); ++i) {
            if (i % 2 == 0) {
                if (s[i] == '0') {
                    cnt1++;
                } else {
                    cnt0++;
                }
            } else {
                if (s[i] == '1') {
                    cnt1++;
                } else {
                    cnt0++;
                }
            }
        }
        return min(cnt0, cnt1);
    }
};

File: daily/rust/1758_min_changes_to_make_alter_binary_string.rs
impl Solution {
    pub fn min_operations(s: String) -> i32 {
        // let mut res = 0;
        // let n = s.len();
        // for i in 0..n {
        //     if s.chars().nth(i).unwrap() as usize - b'0' as usize != i % 2 {
        //         res += 1;
        //     }
        // }
        // std::cmp::min(res, n as i32 - res)

        // let mut res = 0;
        // for (i, c) in s.chars().enumerate() {
        //     if c as usize - b'0' as usize != i % 2 {
        //         res += 1;
        //     }
        // }
        // std::cmp::min(res, s.len() as i32 - res)

        let (mut cnt0, mut cnt1) = (0, 0);
        for i in 0..s.len() {
            if i % 2 == 0 {
                if s.chars().nth(i).unwrap() == '0' {
                    cnt1 += 1;
                } else {
                    cnt0 += 1;
                }
            } else {
                if s.chars().nth(i).unwrap() == '1' {
                    cnt1 += 1;
                } else {
                    cnt0 += 1;
                }
            }
        }
        std::cmp::min(cnt0, cnt1)
    }
}